sdk/src/main/java/com/mapbox/maps/renderer/egl/EGLCore.kt

```diff
@@ -176,16 +179,37 @@ internal class EGLCore(
     return EGL10.EGL_SUCCESS
   }
 
-  /**
-   * Checks for EGL errors.
-   */
-  private fun checkEglErrorNoException(msg: String): Boolean {
+  internal fun addRendererStateListener(listener: RendererSetupErrorListener) {
+    rendererSetupErrorListenerSet.add(listener)
+  }
+
+  internal fun removeRendererStateListener(listener: RendererSetupErrorListener) {
+    rendererSetupErrorListenerSet.remove(listener)
+  }
+
+  internal fun clearRendererStateListeners() {
+    rendererSetupErrorListenerSet.clear()
+  }
+
+  private fun notifyRendererSetupErrorListenersIfNeeded(functionName: String): Int? {
```

Comment:
IMHO for `notify` it's taking too much responsibilities, I'd rename to e.g. `checkEglErrorAndNotify` or split it to several funs

Reply:
fixed